service/frontend/adminHandler.go

```diff
@@ -314,6 +327,239 @@ func (adh *adminHandlerImpl) RemoveTask(
 	return nil
 }
 
+func (adh *adminHandlerImpl) getCorruptWorkflowQueryTemplates(
+	ctx context.Context, request *types.AdminDeleteWorkflowRequest,
+) map[string]func(request *types.AdminDeleteWorkflowRequest) error {
+	client := adh.GetFrontendClient()
+	return map[string]func(request *types.AdminDeleteWorkflowRequest) error{
+		"DescribeWorkflowExecution": func(request *types.AdminDeleteWorkflowRequest) error {
+			_, err := client.DescribeWorkflowExecution(ctx, &types.DescribeWorkflowExecutionRequest{
+				Domain:    request.Domain,
+				Execution: request.Execution,
+			})
+			return err
+		},
+		"GetWorkflowExecutionHistory": func(request *types.AdminDeleteWorkflowRequest) error {
+			_, err := client.GetWorkflowExecutionHistory(ctx, &types.GetWorkflowExecutionHistoryRequest{
+				Domain:    request.Domain,
+				Execution: request.Execution,
+			})
+			return err
+		},
+	}
+}
+
+func (adh *adminHandlerImpl) MaintainCorruptWorkflow(
```

Comment:
Could you explain why `MaintainCorruptWorkflow` is exposed via admin API?
It seems that with the introduction of `DeleteWorkflow` API we should have enough primitives to construct such logic in the CLI alone. Or is it planned to be used in some other way as well?
I'm bringing this up because, IMHO our API surface is already huge. It would be nice if we could contain it to smaller primitive building blocks, that could be used in higher level tools on top.

Reply:
MaintainCorruptWorkflow will have 2 purposes:
- It's used by the corrupt workflow cleaning system workflow
- It's moving the internal dlq-fix logic to open source so the community can use